tests/test_management_api.py

```diff
@@ -99,6 +101,77 @@ def test_get_policy_api(self):
         self.assertTrue(e.has_grouping_policy(["alice", "data2_admin"]))
         self.assertFalse(e.has_grouping_policy(["bob", "data2_admin"]))
 
+    def test_get_policy_matching_function(self):
+        e = self.get_enforcer(
+            get_examples("rbac_with_domain_and_policy_pattern_model.conf"),
+            get_examples("rbac_with_domain_and_policy_pattern_policy.csv"),
+        )
+        e.get_role_manager().add_domain_matching_func(casbin.util.key_match2_func)
+
+        self.assertEqual(
+            e.get_policy(),
+            [
+                ["admin", "domain.*", "data1", "read"],
+                ["user", "domain.1", "data2", "read"],
+                ["user", "domain.1", "data2", "write"],
+            ],
+        )
+
+        self.assertEqual(
+            e.get_filtered_policy(0, "admin"), [["admin", "domain.*", "data1", "read"]]
+        )
+        self.assertEqual(
+            e.get_filtered_policy(0, "user"),
+            [
+                ["user", "domain.1", "data2", "read"],
+                ["user", "domain.1", "data2", "write"],
+            ],
+        )
+
+        self.assertEqual(
+            e.get_filtered_policy(3, "write"),
+            [["user", "domain.1", "data2", "write"]],
+        )
+
+        # Note: "" (empty string) in fieldValues means matching all values.
+        self.assertEqual(
+            e.get_filtered_policy(0, "admin", "", "data1"),
+            [["admin", "domain.*", "data1", "read"]],
+        )
+
+        self.assertEqual(
+            e.get_grouping_policy(),
+            [["alice", "user", "*"], ["bob", "admin", "domain.3"]],
+        )
+        self.assertEqual(
+            e.get_filtered_grouping_policy(0, "alice"), [["alice", "user", "*"]]
+        )
+
+        # check by matching functions
+
+        # but this is wrong use because we have also * domain
+        self.assertEqual(
+            e.get_filtered_grouping_policy(2, "domain.3"),
+            [["bob", "admin", "domain.3"]],
+        )
```

Comment:
These test cases seem to have nothing to do with `matching function`? They have been covered by other tests.

Reply:
Yes, just wanted to make sure that patterns in **policies** does not affect default behaviour. Thought it is not a problem to have additional asserts.
but i can remove them